Lib/json/tool.py

```diff
@@ -5,7 +5,39 @@
 """
 import argparse
 import json
+import re
 import sys
+from _colorize import ANSIColors, can_colorize
+
+
+# The string we are colorizing is a valid JSON,
+# so we can use a looser but simpler regex to match
+# the various parts, most notably strings and numbers,
+# where the regex given by the spec is much more complex.
+_color_pattern = re.compile(r'''
+    (?P<string>"(\\.|[^"\\])*")             |   # String
+    (?P<number>NaN|-?Infinity|[0-9\-+.Ee]+) |   # Number
+    (?P<boolean>true|false)                 |   # Boolean
+    (?P<null>null)                              # Null
+''', re.VERBOSE)
+
+
+def _colorize_json(json_str):
+    colors = {
+        'string': ANSIColors.GREEN,
+        'number': ANSIColors.YELLOW,
+        'boolean': ANSIColors.CYAN,
+        'null': ANSIColors.CYAN,
+    }
+
+    def replace(match):
+        for key in colors:
+            if m := match.group(key):
+                color = colors[key]
+                return f"{color}{m}{ANSIColors.RESET}"
+        return match.group()
+
```

Comment:
Actually, this can even be defined outside (Victor told me that local functions tend to be slower than non-local ones, so when possible, it's better to make them private and outside; let's call it `_replace_match_callback`

Reply:
Good point, I moved them outside.